File: math/strong_number/c++/strong_number.cpp
#include <iostream>

using namespace std;

unsigned factorial(const unsigned num)
{
    if(num <= 1)
    {
        return 1;
    }
    return num*factorial(num-1);
}

bool isStrong(const unsigned num)
{
    if(num == 0) return false;

    unsigned sum = 0;
    for(int i = num; i!=0; i/=10)
    {
        sum += factorial(i%10);
    }
    return sum==num;
}

int main()
{

    cout<<isStrong(145);
    return 0;
}

File: math/strong_number/javascript/strongNumber.js
const factorial = n => (n <= 1) ? 1 : n * factorial(n-1);

/**
 * Checks if a number is equal to the sum of the factorials of its digits.
 *
 * @param num The number to check
 * @returns boolean Whether the number is strong or not.
 */
const checkStrongNum = num => {  
    var sum = 0;
    var temp = num;
    while (temp != 0) {
        digit = temp % 10;
        temp = Math.trunc(temp/10);
        sum += factorial(digit); 
    }
    return (sum == num);
}

module.exports = checkStrongNum;
